beeai/agents/rebase_agent.py

```diff
@@ -39,70 +41,110 @@ class InputSchema(BaseModel):
     dist_git_branch: str = Field(description="dist-git branch to update")
     version: str = Field(description="Version to update to")
     jira_issue: str = Field(description="Jira issue to reference as resolved")
+    build_error: str | None = Field(description="Error encountered during package build")
 
 
 class OutputSchema(BaseModel):
     success: bool = Field(description="Whether the rebase was successfully completed")
     status: str = Field(description="Rebase status")
+    srpm_path: Path | None = Field(description="Absolute path to generated SRPM")
     error: str | None = Field(description="Specific details about an error")
 
 
+def get_instructions() -> str:
+    return """
+      You are an expert on rebasing packages in RHEL ecosystem.
+
+      To rebase package <PACKAGE> to version <VERSION> in dist-git branch <DIST_GIT_BRANCH>, do the following:
+
+      1. Check if the current version is older than <VERSION>. To get the current version,
+         you can use `rpmspec -q --queryformat "%{VERSION}\n" --srpm <PACKAGE>.spec`.
+         To compare versions, use `rpmdev-vercmp`. If the current version is not older than <VERSION>,
+         rebasing doesn't make sense, so end the process with an error.
+
+      2. Try to find past rebases in git history to see how this particular package does rebases.
+         Keep in mind what parts of the spec file are usually changed. At the minimum a rebase should
+         change `Version` and `Release` tags (or corresponding macros) and add a new changelog entry,
+         but sometimes other things are changed - if that's the case, try to understand the logic behind it.
```

Comment:
I'm curious if this could potentially help with cases like LLVM rebase, but I expect there need to be more explicit instructions for that